File: Find_the_Middle_Index_in_Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMiddleIndex = function(nums) {
    let left = 0;
    let right = nums.reduce((acc, curr) => acc + curr, 0);

    for (let i = 0; i < nums.length; i++) {
      right -= nums[i];

      if (left === right) {
        return i;
      }

      left += nums[i];
    }

    return -1;    
};

File: Find_the_Middle_Index_in_Array.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        int left = 0;
        int right = 0;
        for (int num : nums) {
            right += num;
        }
        
        for (int i = 0; i < nums.length; i++) {
            right -= nums[i];
            
            if (left == right) {
                return i;
            }
            
            left += nums[i];
        }
        
        return -1;        
    }
}
